src/index.js

```diff
@@ -267,21 +267,19 @@ class PubsubBaseProtocol extends EventEmitter {
 
   /**
    * Validates the given message. The signature will be checked for authenticity.
+   * Throws an error on invalid messages
    * @param {InMessage} message
-   * @returns {Promise<Boolean>}
+   * @returns {Promise<void>}
    */
   async validate (message) { // eslint-disable-line require-await
     // If strict signing is on and we have no signature, abort
     if (this.strictSigning && !message.signature) {
-      this.log('Signing required and no signature was present, dropping message:', message)
-      return false
+      throw errcode(new Error('Signing required and no signature was present'), 'ERR_MISSING_SIGNATURE')
```

Comment:
We will need to check this codes in the router? If so, we should export the errors.
I recommend having something like: https://github.com/libp2p/js-libp2p/blob/master/src/errors.js and export them to be used by the routers